lib/elixir/test/elixir/kernel/expansion_test.exs

```diff
@@ -957,157 +957,6 @@ defmodule Kernel.ExpansionTest do
     end
   end
 
-  describe "cond" do
-    test "expands each clause" do
-      before_expansion =
-        quote do
-          cond do
-            x = 1 -> x
-            true -> x
-          end
-        end
-
-      after_expansion =
-        quote do
-          cond do
-            x = 1 -> x
-            true -> x()
```

Comment:
Actually, there is one way, but it's a bit strange. It would be to put the case inside of a try and throw and catch the value and do the nesting inside of the catch.

```
try do
  value = x = 1

  (value == nil or value == false) and throw(:cond)
  x
catch
  :cond ->
    try do
      value = true

      (value == nil or value == false) and throw(:cond)
      x
    catch
      :cond ->
        raise CondClauseError
    end
end
```